src/main/resources/templates/admin/category/add.html

```diff
@@ -0,0 +1,103 @@
+                                                    <i class="fas fa-search"></i>
+                                                </button>
+                                            </div>
+                                        </div>
+                                    </div>
+                                </div>
+                                -->
+                                <!-- /.card-header -->
+                                <!-- general form elements -->
+                                <div class="card card-primary">
+                                    <div class="card-header">
+                                        <h3 class="card-title">Add Category</h3>
+                                    </div>
+                                    <!--<form th:action="@{/categories}" method="post">
+                                        <label for="categoryName">Category</label>
+                                        <input type="text" id="categoryName" name="categoryName" required>
+                                        <button type="submit">Add Category</button>
+                                    </form>
+                                    -->
+                                    <!-- /.card-header -->
+                                    <div class="card-body">
+                                        <div class="form-group">
+                                            <label for="categoryName">Name:</label>
+                                            <input th:field="*{name}" type="text" class="form-control" id="categoryName" placeholder="Category name">
+                                            <input th:field="*{id}" type="hidden" />
+                                            <span class="error" th:if="${message}" th:text="${message}" th:class="${message}" />
+                                            <span class="error" th:if="${#fields.hasErrors('name')}" th:errors="*{name}" style="color:red">Name Error</span>
+                                            <span th:text="${sameCategory}" style="color:red"></span>
+                                        </div>
+                                    </div>
+                                    <!-- /.card-body -->
+                                    <div class="card-footer">
+                                        <button type="submit" class="btn btn-primary">Add</button>
+
+                                    </div>
+                                </div>
+                                <!-- /.card -->
+                                <!-- /.card-body -->
+                            <!--</div> -->
+                            <!-- /.card -->
+                       <!-- </div>
+                    </div> -->
+                </form>
+                <!-- /.row -->
+            </div><!-- /.container-fluid -->
+        </div>
+        <!-- /.content -->
+    </div>
+    <!-- /.content-wrapper -->
+
+    <th:block th:replace="admin/template :: rightbar" />
```

Comment:
Use the complete syntax of fragment expressions
```suggestion
    <th:block th:replace="~{admin/template :: rightbar}" />
```